src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 6 : 0)
```

Comment:
> The subtraction method looks more readable to me.

Can't account for taste.

> logically correct

Actually, it's illegal for both to be XRP, so you can do basically `if (isXRP(gets)) { ... } else if (isXRP(pays)) { ... } else  {...}`